model/task_lifecycle.go

```diff
@@ -1393,7 +1396,40 @@ func doRestartFailedTasks(tasks []string, user string, results RestartResults) R
 	return results
 }
 
-func ClearAndResetStrandedTask(h *host.Host) error {
+// ClearAndResetStrandedContainerTask clears the container task dispatched to a
+// pod. It also resets the task so that the current task execution is marked as
+// finished and, if necessary, a new execution is created to restart the task.
+// TODO (PM-2618): should probably block single host task groups once they're
+// supported.
+func ClearAndResetStrandedContainerTask(p *pod.Pod) error {
+	runningTaskID := p.RunningTask
```

Comment:
Since we never reassign it maybe it's best to just directly use `p.RunningTask` instead of declaring a new variable

Reply:
`(*pod.Pod).ClearRunningTask` unsets the running task but we still need it to find the task itself.